examples/Basics.ipynb

```diff
@@ -0,0 +1,480 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Basics of draive\n",
+    "\n",
+    "Draive is a framework focused on building high quality applications utilizing various machine learning models, especially LLMs. It provides a set of tools allowing to easily manage data flows, dependencies and state propagation across the application. Draive is suitable for building servers, workers and local solutions including CLI and GUI based. Core principles of the framework are strongly connected to the functional programming and structured concurrency concepts. Basic building blocks are asynchronous functions which can be easily composed and annotated to serve various roles. To ensure proper state management draive comes with the concept of contextual state propagation wrapped around python context managers. Various configurations can be deeply propagated through the call tree to ensure state consistency. In order to avoid unexpected mutations during the execution and since we leverage concepts of functional programming draive comes with the base data classes which limit state mutability forcing to prepare local copies of mutated data. Additionally, basing on the structured concurrency concepts draive allows to propagate context states to concurrently executed functions. Since state mutations are already limited and controlled this allows easy and safe parallelization of almost all operations. Finally all of the draive code is fully typed as strictly as possible making as much use of the static analysis as possible. This means that a lot of potential issues can be catch before running the code. It is strongly recommended to use strict, full type linting if possible. High quality is not only the phrase we are taking it seriously!"
```

Comment:
> utilizing various machine learning models, especially LLMs

Nope, I wouldn't use it to run inference on discriminative models. It's solely dedicated to LLM/LMM and external APIs when you don't host your own model.

Reply:
There's too much text/content in the intro delving into details. w/o examples.

Reply:
Please run this description against grammarly ;p